### PR TITLE
Run ExtractWidgetAction in transaction.

### PR DESCRIPTION
**Pull request by scheglov**
https://github.com/flutter/flutter-intellij/issues/2165

### CODE REVIEW

File: src/io/flutter/actions/ExtractWidgetAction.java [hunk 1]
```diff
@@ -38,6 +38,11 @@ public void actionPerformed(AnActionEvent event) {
       new ExtractWidgetDialog(project, editor, refactoring).show();
     }
   }
+
+  @Override
+  public boolean startInTransaction() {
+    return true;
+  }
```
Comment (pq) [inline src/io/flutter/actions/ExtractWidgetAction.java]: Well that was easy! 🎉

Comment (scheglov) [reply src/io/flutter/actions/ExtractWidgetAction.java]: :-D

**Comment by devoncarew** [comment]
@scheglov, is this fix worth back-porting to the current release?

**Comment by scheglov** [comment]
If we are going to do any back-porting anyway, we might add it.
Otherwise it is not worth it.

I think this happens only when errors are logged.
Otherwise it is benign.

We compute refactoring in Analysis Server, so we probably don't actually need this transaction.

**Pull request merged**